cmd/ci-secret-bootstrap/main.go

```diff
@@ -480,30 +628,54 @@ func main() {
 			logrus.WithError(err).Fatal("Failed to logout.")
 		}
 	})
-	defer logrus.Exit(0)
+
 	ctx := context.TODO()
 	var errs []error
 	// errors returned by constructSecrets will be handled once the rest of the secrets have been uploaded
 	secretsMap, err := constructSecrets(ctx, o.config, bwClient, o.maxConcurrency)
 	if err != nil {
 		errs = append(errs, err)
 	}
-	if o.dryRun {
-		tmpFile, err := ioutil.TempFile("", "ci-secret-bootstrapper")
-		if err != nil {
-			logrus.WithError(err).Fatal("failed to create tempfile")
-		}
-		defer tmpFile.Close()
-		logrus.Infof("Dry-Run enabled, writing secrets to %s", tmpFile.Name())
-		if err := writeSecrets(secretsMap, tmpFile); err != nil {
-			errs = append(errs, fmt.Errorf("failed to write secrets on dry run: %w", err))
+
+	// if validate-bitwarden-items-usage is specified we only want to check if all the
+	// bitwarden items are being used in the given configuration.
```

Comment:
Should we make a new binary for this function since there is no logic overlapping with the existing tool?

Reply:
Well this tool does already what we need here. Also the config file belongs here so it makes more sense to make the tool validate its own config

Reply:
Ack. You can mark this as resolved.